compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -1595,10 +1595,22 @@ fn evaluate_prefix_with_value(rhs: Value, operator: UnaryOp, location: Location)
     match operator {
         UnaryOp::Minus => match rhs {
             Value::Field(value) => Ok(Value::Field(-value)),
-            Value::I8(value) => Ok(Value::I8(-value)),
-            Value::I16(value) => Ok(Value::I16(-value)),
-            Value::I32(value) => Ok(Value::I32(-value)),
-            Value::I64(value) => Ok(Value::I64(-value)),
+            Value::I8(value) => value
+                .checked_neg()
+                .map(Value::I8)
+                .ok_or_else(|| InterpreterError::PrefixMathError { location }),
+            Value::I16(value) => value
+                .checked_neg()
+                .map(Value::I16)
+                .ok_or_else(|| InterpreterError::PrefixMathError { location }),
+            Value::I32(value) => value
+                .checked_neg()
+                .map(Value::I32)
+                .ok_or_else(|| InterpreterError::PrefixMathError { location }),
+            Value::I64(value) => value
+                .checked_neg()
+                .map(Value::I64)
+                .ok_or_else(|| InterpreterError::PrefixMathError { location }),
             Value::U8(value) => Ok(Value::U8(0 - value)),
```

Comment:
Do we allow unary negation for unsigned integers?

Reply:
Hm, good point! Apparently it's forbidden in non-comptime code (it gives an error) but in comptime code it crashes the compiler:

```noir
fn main() {
    comptime {
        let i: u8 = 1;
        let _ = -i;
    }
}
```

```
The application panicked (crashed).
Message:  attempt to subtract with overflow
Location: compiler/noirc_frontend/src/hir/comptime/interpreter.rs:1614
```

Maybe the error would still be reported later on but it ends up being evaluated at comptime 🤔 

I'll try to fix it in this PR too.

Reply:
I made it error. The only issue is that we now get the "same" error twice: once when type-checking, another one when evaluating comptime code. I thought about not erroring and just going on with the same value, but then it could be confusing to continue with an incorrect value if that incorrect value leads to more errors later on.

But then, if you have this code:

```noir
    comptime {
        let i: u8 = 1 + 300;
    }
```

you also get two errors on "300". The messages are slightly different, but they are essentially the same error. So maybe double erroring is fine (or at least it's not a new thing of this PR).